include/appfwk/ModuleConfiguration.hpp

```diff
@@ -21,43 +21,39 @@
 
 namespace dunedaq {
 
-ERS_DECLARE_ISSUE(appfwk,                                                   ///< Namespace
-                  NotADaqModule,                                            ///< Issue class name
+ERS_DECLARE_ISSUE(appfwk,                                                                  ///< Namespace
+                  NotADaqModule,                                                           ///< Issue class name
                   "Application contains a resource " << res << " that is not a DaqModule", ///< Message
-                  ((std::string)res)                                      ///< Message parameters
+                  ((std::string)res)                                                       ///< Message parameters
 )
 
 namespace coredal {
-  class DaqModule;
-  class Session;
-  class Application;
+class DaqModule;
+class Session;
+class Application;
 }
 namespace appfwk {
 
-  class ModuleConfiguration {
-    std::vector<const dunedaq::coredal::DaqModule*> m_modules;
-    iomanager::Queues_t m_queues;
-    iomanager::Connections_t m_networkconnections;
-    static std::shared_ptr<ModuleConfiguration> s_instance;
-    ModuleConfiguration(){}
-  public:
-    void initialise();
-    static std::shared_ptr<ModuleConfiguration> get() {
-      if (!s_instance) {
-        s_instance.reset(new ModuleConfiguration);
-      }
-      return s_instance;
-    }
-    const iomanager::Queues_t& queues() {return m_queues;}
-    const iomanager::Connections_t& networkconnections() {
-      return m_networkconnections;
-    }
-    const std::vector<const coredal::DaqModule*>& modules() {return m_modules;}
-
-    template<typename T> const T* module(const std::string& name){
-      return ConfigurationManager::get()->get_dal<T>(name);
-    }
-  };
+class ModuleConfiguration
+{
+  std::vector<const dunedaq::coredal::DaqModule*> m_modules;
+  iomanager::Queues_t m_queues;
+  iomanager::Connections_t m_networkconnections;
+  std::shared_ptr<ConfigurationManager> m_config_mgr;
+
+public:
+  ModuleConfiguration(std::shared_ptr<ConfigurationManager> mgr);
```

Comment:
Should this be explicit? I seem to remember being told that constructors with a single argument should be explicit but I can't remember why.

Reply:
It disables implicit conversions, which is generally a good thing.